File: Quasics2024Code/Crescendo-XrpEnabled/src/main/cpp/commands/RunShooter.cpp
// Copyright (c) 2024 Quasics, FIRST, and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "commands/RunShooter.h"

RunShooter::RunShooter(Shooter &shooter, double shooterSpeed, bool shooting)
    : m_shooter(shooter),
      m_shooterSpeed(shooting ? std::abs(shooterSpeed)
                              : -std::abs(shooterSpeed)),
      m_shooting(shooting) {
  AddRequirements(&m_shooter);
}

// Called when the command is initially scheduled.
void RunShooter::Initialize() {
  m_shooter.SetFlywheelSpeed(m_shooterSpeed);
}

// Called repeatedly when this Command is scheduled to run
void RunShooter::Execute() {
  m_shooter.SetFlywheelSpeed(m_shooterSpeed);
}

// Called once the command ends or is interrupted.
void RunShooter::End(bool interrupted) {
  m_shooter.Stop();
}

// Returns true when the command should end.
bool RunShooter::IsFinished() {
  return false;
}

File: Quasics2026Code/RebuiltRobot/src/main/java/frc/robot/commands/RunShooter.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot.commands;

import edu.wpi.first.wpilibj2.command.Command;
import frc.robot.subsystems.real.RealShooter;

/* You should consider using the more terse Command factories API instead https://docs.wpilib.org/en/stable/docs/software/commandbased/organizing-command-based.html#defining-commands */
public class RunShooter extends Command {
  RealShooter m_shooter;
  private double m_shooterSpeed;
  private double m_kickerSpeed;

  /** Creates a new RunShooter. */
  public RunShooter(RealShooter shooter, double shooterSpeed, double kickerSpeed, boolean shooting) {
    m_shooter = shooter;
    m_shooterSpeed = shooterSpeed;
    m_kickerSpeed = kickerSpeed;
    if (shooting) {
      m_shooterSpeed = Math.abs(shooterSpeed);
      m_kickerSpeed = Math.abs(kickerSpeed);
    } else {
      m_shooterSpeed = -Math.abs(shooterSpeed);
      m_kickerSpeed = -Math.abs(kickerSpeed);
    }

    // Use addRequirements() here to declare subsystem dependencies.
    addRequirements(shooter);
  }

  // Called when the command is initially scheduled.
  @Override
  public void initialize() {
    m_shooter.setFlywheelSpeed(m_shooterSpeed);
    m_shooter.setKickerSpeed(m_kickerSpeed);
  }

  // Called every time the scheduler runs while the command is scheduled.
  @Override
  public void execute() {
    m_shooter.setFlywheelSpeed(m_shooterSpeed);
    m_shooter.setKickerSpeed(m_kickerSpeed);
  }

  // Called once the command ends or is interrupted.
  @Override
  public void end(boolean interrupted) {
    m_shooter.stop();
  }
}
